src/client/mod.rs

```diff
@@ -123,8 +126,41 @@ impl<T: Transport, L: LoginCredentials> TwitchIRCClient<T, L> {
     /// it will not be cut short or split into multiple messages (what happens is determined
     /// by the behaviour of the Twitch IRC server).
     pub async fn say(&self, channel_login: String, message: String) -> Result<(), Error<T, L>> {
-        // The prefixed "." prevents execution of commands
-        self.privmsg(channel_login, format!(". {}", message)).await
+        self.say_in_response(channel_login, message, None).await
+    }
+
+    /// Say a chat message in the given Twitch channel, but send it as a response to another message if `reply_to_id` is specified.
+    ///
+    /// Behaves the same as `say()` when `reply_to_id` is None, but tags the original message and it's sender if specified.
+    pub async fn say_in_response(
+        &self,
+        channel_login: String,
+        message: String,
+        reply_to_id: Option<String>,
+    ) -> Result<(), Error<T, L>> {
+        let mut tags = IRCTags::new();
+
+        if let Some(id) = reply_to_id {
+            tags.0.insert("reply-parent-msg-id".to_string(), Some(id));
+        }
+
+        let irc_message = IRCMessage::new(
+            tags,
+            None,
+            "PRIVMSG".to_string(),
+            vec![format!("#{}", channel_login), format!(". {}", message)], // The prefixed "." prevents commands from being executed
+        );
+        self.send_message(irc_message).await
+    }
+
+    /// Replies to a given Privmsg, tagging the original message and it's sender.
+    pub async fn reply_to_privmsg(
+        &self,
+        message: String,
+        reply_to: PrivmsgMessage,
```

Comment:
you should borrow the PrivmsgMessage, i.e. the parameter should be of type `&PrivmsgMessage`, not `PrivmsgMessage`

Reply:
In that case, either the parameters in `say`, `say_in_response` and `privmsg` would have to be changed to be references (which would be a breaking change), or `PrivmsgMessage`'s fields would have to be cloned internally. I'm not sure what would be the better solution.

Reply:
I think cloning the strings is acceptable.